projects/frontend/data-pipelines/gui/e2e/plugins/helpers/job-helpers.plugins.js

```diff
@@ -359,7 +359,9 @@ const provideDataJobsExecutions = (taskConfig, config) => {
                                                             jobExecutionTimeout,
                                                             prevCommandInner.lastDeployment,
                                                             targetExecutions -
-                                                                prevCommandInner.executions,
+                                                                prevCommandInner
+                                                                    .executions
```

Comment:
It's trying to cast the second operand to integer and result is
* if array is empty then casting `-[]` will result to `-0` which is identical to `0`
  * expression would look like `targetExecutions - (-0)` which is identical to `targetExecutions`
* if array is not empty, has elements inside behavior is unexpected, casting `-[1,2,3]` will result to `NaN`
  * expression would look like `targetExecutions - NaN` which results to `NaN`

I noticed the problem while debugging yesterday after we finished our call.
Such problems are not found during compile time, because Cypress custom plugins are written in vanilla JavaScript and there is not strongly typing.

Reply:
Errors in vanilla JavaScript are very rare because of types mismatch, instead interpreter try to cast operands in same type and provide end result from an expression, and because of it Typescript was born like superscript of the JavaScript with strongly type